From 8723a7603d98fa08039e21ee9799d5293808c96f Mon Sep 17 00:00:00 2001
From: Rishi Gupta <rishgupt@codeaurora.org>
Date: Sun, 23 Aug 2020 20:10:45 +0530
Subject: [PATCH 1/3] ubifs: Populate UUID in superblock of ubi file system

Currently the super block of ubi file system does not have
uuid in super block structure. UUID is required by integrity
measurement architecture driver for deciding whether files
in given partition are to be appraised or not based on uuid
matching.

Change-Id: I6e5036cb6c8a908024bc80520d042437c657997a
Signed-off-by: Rishi Gupta <rishgupt@codeaurora.org>
---
 fs/ubifs/super.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/fs/ubifs/super.c b/fs/ubifs/super.c
index 76f1873..91a2b32 100644
--- a/fs/ubifs/super.c
+++ b/fs/ubifs/super.c
@@ -2093,6 +2093,9 @@ static int ubifs_fill_super(struct super_block *sb, void *data, int silent)
 		goto out_umount;
 	}
 
+	/* populate uuid into superblock structure */
+	memcpy(&sb->s_uuid, &c->uuid, sizeof(uuid_t));
+
 	mutex_unlock(&c->umount_mutex);
 	return 0;
 
-- 
2.7.4

